pipecat-client-android/src/main/java/ai/pipecat/client/PipecatClient.kt

```diff
@@ -0,0 +1,532 @@
+                    }
+
+                    MsgServerToClient.Type.BotLlmText -> {
+                        val data: MsgServerToClient.Data.BotLLMTextData =
+                            JSON_INSTANCE.decodeFromJsonElement(msg.data)
+
+                        callbacks.onBotLLMText(data)
+                    }
+
+                    MsgServerToClient.Type.BotTtsText -> {
+                        val data: MsgServerToClient.Data.BotTTSTextData =
+                            JSON_INSTANCE.decodeFromJsonElement(msg.data)
+
+                        callbacks.onBotTTSText(data)
+                    }
+
+                    MsgServerToClient.Type.BotLlmStarted -> callbacks.onBotLLMStarted()
+                    MsgServerToClient.Type.BotLlmStopped -> callbacks.onBotLLMStopped()
+
+                    MsgServerToClient.Type.BotTtsStarted -> callbacks.onBotTTSStarted()
+                    MsgServerToClient.Type.BotTtsStopped -> callbacks.onBotTTSStopped()
+
+                    MsgServerToClient.Type.ServerMessage -> {
+                        callbacks.onServerMessage(JSON_INSTANCE.decodeFromJsonElement(msg.data))
+                    }
+
+                    MsgServerToClient.Type.Metrics -> {
+                        callbacks.onMetrics(JSON_INSTANCE.decodeFromJsonElement(msg.data))
+                    }
+
+                    MsgServerToClient.Type.LlmFunctionCall -> {
+
+                        val functionCallData =
+                            JSON_INSTANCE.decodeFromJsonElement<LLMFunctionCallData>(msg.data)
+
+                        val handler = functionCallHandlers[functionCallData.functionName]
+
+                        val activeConnection = connection
+
+                        val resultHandler: (Value) -> Unit = { resultData ->
+
+                            thread.runOnThread {
+                                if (activeConnection == connection) {
+                                    sendMessage(
+                                        MsgClientToServer.LlmFunctionCallResult(
+                                            msgId = msg.id ?: UUID.randomUUID().toString(),
+                                            data = LLMFunctionCallResult(
+                                                functionName = functionCallData.functionName,
+                                                toolCallID = functionCallData.toolCallID,
+                                                arguments = functionCallData.args,
+                                                result = JSON_INSTANCE.encodeToJsonElement(
+                                                    resultData
+                                                )
+                                            )
+                                        )
+                                    )
+                                }
+                            }
+                        }
+
+                        if (handler == null) {
```

Comment:
per the other discussion, i do not believe this is an if/else situation.

Separately, I don't understand Kotlin. How does the registered functionCallHandler pass back the resultData used in the resultHandler?

Also note that in client-js/ios, I don't believe we are providing built-in support for sending back a result for the generic `onLLMFunctionCall` callback.